fieldservice_stock/models/fsm_order.py

```diff
@@ -35,15 +35,27 @@ def _default_warehouse_id(self):
                                    required=True, readonly=True,
                                    default=_default_warehouse_id,
                                    help="Warehouse used to ship the materials")
+    # FSM Order return
+    return_ids = fields.One2many(
+        'fsm.order.return', 'order_id', string="Return Lines", )
+    return_count = fields.Integer(string='Return Orders',
+                                  compute='_compute_picking_ids')
 
     @api.depends('picking_ids')
     def _compute_picking_ids(self):
         for order in self:
-            order.delivery_count = len(order.picking_ids)
+            order.delivery_count = len(
+                [picking for picking in order.picking_ids if
+                 picking.picking_type_id.code == 'outgoing'])
+            order.return_count = len(
+                [picking for picking in order.picking_ids if
+                 picking.picking_type_id.code == 'incoming'])
 
     def action_confirm(self):
         if self.location_id and self.line_ids and self.warehouse_id:
             self.line_ids._confirm_picking()
+            if self.return_ids:
+                self.return_ids._confirm_picking()
```

Comment:
@smangukiya It doesn't work if I have a return line (inbound) and no line (outbound). I keep getting the error message below. Confirming should confirm the draft lines and draft returns if there is any.

Reply:
@smangukiya I fixed the logic but we still need to filter the lines and returns with draft ones.